### ISSUE TITLE
factory: adversarial review and oracle hardening of synthetic SWE tasks with a rights-cleared reviewer

### ISSUE THREAD

**Issue by rmems** [rmems/synthetic-factory#158]
## Planning / research attribution

**GPT-5.6-Sol (OpenAI / ChatGPT), 2026-08-28**, with Raul Montoya Cardenas directing the adversarial review/oracle-hardening stage.

## Parent

#153 — synthetic SWE pipeline utility / no-degradation program.

Feeds #155 — verified software-engineering synthetic release candidate.

## Goal

Use Meta Muse Spark 1.2 as a **review/test-critic component** that tries to falsify weak synthetic tasks, patches, tests, and oracle assumptions before records can enter the curated SWE release.

This issue does **not** rank Muse against Fable or GPT as a data generator. Its purpose is to strengthen the pipeline's evidence quality so the final research question can be answered downstream: does the resulting synthetic training data help a model without unacceptable degradation?

## Review surfaces

Muse may review:

- task/specification consistency;
- candidate reference implementation;
- solver patch;
- hidden/property/metamorphic tests;
- mutation survivors;
- regression risk;
- contradictory requirements;
- insufficient validation evidence;
- architecture/ownership mistakes when a deterministic contract exists.

### `defect_review`

```text
task + repository snapshot
+ candidate patch
+ visible validation evidence
→ review findings
→ severity / location / evidence
→ adjudication
```

### `adversarial_test_design`

```text
specification
+ candidate implementation
→ proposed discriminating tests / mutants
→ execution results
→ revised test plan if needed
```

### `requirements_clarification`

```text
underspecified or contradictory task
→ minimum missing decision set
→ resolved / quarantined task status
```

These may remain internal pipeline artifacts or map into #155 record kinds where they provide useful observable training signal.

## Reviewer truth boundary

Muse does not self-label its findings as true.

Adjudicate review findings using evidence such as:

- seeded defect inventory;
- compiler/test execution;
- mutation testing;
- reference behavior;
- property/metamorphic tests;
- source contract/specification;
- bounded human adjudication where semantics remain ambiguous.

Unsupported findings remain reviewer false positives; do not convert confident prose into truth.

## Oracle-hardening objective

The review stage should answer:

> Can an important wrong implementation still pass the current oracle?

If yes, the task stays `provisional` or `invalid` until the oracle is strengthened.

Example loop:

```text
candidate task/oracle
→ mutate / adversarially challenge
→ surviving incorrect behavior?
    yes → strengthen tests / quarantine
    no  → retain evidence
→ independent oracle validation
```

## Provenance

Use #154 with Muse recorded as `reviewer`. Keep task-author, solver, reviewer, oracle, and curation roles separate.

## Required metrics

Report where ground truth permits:

```text
review true positives
review false positives
review false negatives
precision / recall
severity calibration
mutation kill rate before review
mutation kill rate after hardening
tasks rejected/quarantined because oracle was weak
accepted records after curation
```

Reviewer verbosity is not a quality metric.

## Relationships

- **Parent:** #153
- **Feeds / hardens:** #155
- **Blocked by:** #32, #154
- **Related solver stage:** #157
- **Task-author stage:** #156

## Acceptance criteria

- [ ] Review findings are adjudicated against external evidence rather than Muse confidence.
- [ ] Reviewer identity is distinct from task author, solver, and oracle.
- [ ] Weak or contradictory task/oracle pairs can be quarantined explicitly.
- [ ] At least one oracle-hardening path exercises mutation/adversarial evidence where practical.
- [ ] False-positive review findings remain observable errors.
- [ ] Final output feeds the single #155 release candidate rather than requiring a Muse-specific HF dataset repository.
- [ ] Hidden/internal reasoning does not enter training-ready views.

## Tracking / Linear sync

GitHub is source of truth. Let GitHub→Linear create the twin automatically. After sync, set the #153 twin as parent, use the existing Synthetic Data Factory project / Factory v0.2.0 milestone, and preserve relations to #155/#156/#157.

**Comment by rmems** [comment]
Retitled 2026-09-03 under #161: reviewer verdicts that rewrite or harden tasks are derivative of the reviewer's output, so the reviewer role goes to a rights-cleared model; Muse Spark 1.2 stays a bounded research comparator. The old title is in the edit history.

**Comment by rmems** [comment]
**DELETE** — recommendation from **01 Synthetic Factory Debt Closer** (Tech Debt Mill).

Debt-mode freeze: no new train/gen, VSET–SWE campaign, or neuromorphic dataset-family work. Close unless Raul explicitly reopens for a named mill.

**Comment by rmems** [comment]
**Status update 2026-09-04**

Retitled to a rights-cleared reviewer; no implementation. Depends on #170/#163 for the reviewer model rows. Muse Spark 1.2 stays a research comparator; review verdicts that rewrite tasks would inherit the reviewer's rights status, which is why the role moved.

— Claude Fable 5.1 (Claude Code)
